packages/forms/src/model/abstract_model.ts

```diff
@@ -401,7 +401,7 @@ export abstract class AbstractControl<TValue = any, TRawValue extends TValue = T
    *
    * @internal
    */
-  private _rawAsyncValidators: AsyncValidatorFn|AsyncValidatorFn[]|null;
+  private _rawAsyncValidators!: AsyncValidatorFn|AsyncValidatorFn[]|null;
```

Comment:
Should the type be:
```suggestion
  private _rawAsyncValidators: ReadonlyArray<AsyncValidatorFn> = [];
```

Always an array and always readonly

Reply:
The array can be mutated later on by the `addValidators` and `removeValidators` methods. We just don't watch to mutate the original array given to us by the author.

Reply:
> Always an array and always readonly

There is a code path where `_rawValidators` and `_rawAsyncValidators` may become a function:

https://github.com/angular/angular/blob/main/packages/forms/src/model/abstract_model.ts#L441-L446

However, we can do further refactoring (in a followup PR if needed) to replace the implementation of the `validator` and `asyncValidator` setters with `setValidators` and `setAsyncValidators` calls, in which case the `_rawValidators` and `_rawAsyncValidators` would always be arrays.

Reply:
I don't think that this code path is relevant for the fix here though. We only need to clone the value when it's being set to an array and the `validator` getter/setter is a `ValidatorFn|null`.